internal/configs/resource.go

```diff
@@ -49,6 +50,14 @@ type Resource struct {
 	// If this is nil, then this resource is essentially public.
 	Container Container
 
+	// IsOverridden indicates if the resource is being overriden. It's used in
+	// testing framework to not call the underlying provider.
+	IsOverridden bool
```

Comment:
Yep, it could be done this way. We will need to check for a nil map value (not overridden), a zero-length map (overridden with no values), and a filled map (overridden with values). I just find it harder to understand that way from the reader's perspective.

Also, I think it's easy to misuse nil and zero-length maps this way so an additional flag comes in handy.

Reply:
that makes sense... just wanted to know the idea behind this.
Great work Thanks!